src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -315,25 +315,22 @@ public override object Clone()
             {
                 //inv.type |= nFetchFlags;
 
-                // TODO: This is incorrect, in blocks only mode we should just add to known inventory but not relay
-                if (this.isBlocksOnlyMode)                 
-                    this.logger.LogInformation("Transaction ID '{0}' inventory sent in violation of protocol peer '{1}'.", inv.Hash, peer.RemoteSocketEndpoint);
+                lock (this.lockObject)
```

Comment:
I think in general, it is better to grab the lock only once as it was in the original code instead of grabbing it for each item and then release it again.